src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerServiceTest.java

```diff
@@ -195,6 +209,29 @@ private GeneralPartnerDao createDao() {
         return dao;
     }
 
+    private List<GeneralPartnerDao> buildGeneralPartnerDaoList() {
+        List<GeneralPartnerDao> daoList = new ArrayList<>();
+        GeneralPartnerDao dao1 = createDao();
+        setSelfLinkOnDaoWithTransactionId(dao1, TRANSACTION_ID);
+        daoList.add(dao1);
+        GeneralPartnerDao dao2 = createDao();
+        setSelfLinkOnDaoWithTransactionId(dao2, "wrong1");
+        daoList.add(dao2);
+        GeneralPartnerDao dao3 = createDao();
+        setSelfLinkOnDaoWithTransactionId(dao3, "wrong2");
+        daoList.add(dao3);
+        GeneralPartnerDao dao4 = createDao();
+        setSelfLinkOnDaoWithTransactionId(dao4, TRANSACTION_ID);
+        daoList.add(dao4);
+        return daoList;
```

Comment:
Related to an earlier comment. Don't need to check the self links for the `transaction_id` as it's a separate field.